### DISCUSSION TITLE
Shell integration

### DISCUSSION THREAD

**Discussion by 0x241F31** [rhaiscript/rhai#852] [Ideas]
I was wondering if Rust-like scripting language could be used to make shell scripts? [There is an example](https://rhai.rs/book/engine/hello-world.html?highlight=sheban#admonition-tip-unix-shebangs), but maybe it's worth to have tooling around shell like [zx](github.com/google/zx) or [bun shell](https://bun.sh/docs/runtime/shell#reading-output-line-by-line) does?

Here is example of using xz: 
```js
function swap-outputs {
  zx << 'EOF'
    let monitors = await $`wlr-randr --json`.then(JSON.parse)
    if (monitors.length < 2) {
      echo`less then 2 monitors connected`
      process.exit(1)
    }

    if (monitors.every(out => out.enabled)) {
      let [out1, out2] = monitors;
      await $`wlr-randr --output ${out1.name} --off`
      await $`wlr-randr --output ${out2.name} --on`
      process.exit(0)
    }
    let tasks = monitors.map(out => $`wlr-randr --output ${out.name} ${out.enabled ? '--off' : '--on'}`)
    await Promise.allSettled(tasks)
EOF
}
```

How similiar script will look like in rhai?

**Comment by schungx** [comment]
Sorry I just realize that this question is here and hasn't been answered.

I suppose you _can_, but you may not want to.  Why use Rhai when you have lots of different scripting languages for different shells?  Perl, awk, bash, csh, etc.  The list goes on and on.  Why use Rhai?

**Reply by elenapan** [reply]
Hi, I believe I can describe a relevant use case, albeit for a specific project that uses Rhai.

I am using [yolk](https://github.com/elkowar/yolk/), a dotfile manager which uses Rhai for configuration and templating.

I would like to spawn processes through Rhai and use their output or exit code to set variables which will be used in the yolk templating system.

For example:
1. Run `free --giga | awk '/^Mem: / {print $2}'` to get the system's total RAM in Gigabytes.
2. Use this value in a yolk template to configure certain system/app settings accordingly.

**Reply by schungx** [reply]
I suppose there isn't any reason why you can't do it, but constructing a command line with arguments, syntax-wise, may not be as elegant and pretty in Rhai code vs some of other scripting systems.

**Comment by schungx** [comment]
Closing this for now. Feel free to reopen if necessary.
